google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java

```diff
@@ -186,7 +186,9 @@ public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset)
     // Handle schema updates in a Thread-safe way by locking down the operation
     synchronized (this) {
       // Create a new stream writer internally if a new updated schema is reported from backend.
-      if (this.streamWriter.getUpdatedSchema() != null) {
+      TableSchema updatedSchema = this.streamWriter.getUpdatedSchema();
+      // Only refresh when the reference of the schema is different.
+      if (updatedSchema != null && this.tableSchema != updatedSchema) {
```

Comment:
Seems this is just for the extreme case of if a schema update event happens within 10ms of Stream creation. I think we should be OK without this comparison.

Reply:
Discussed offline, changed to change test only